File: python/questao11.py
def calcula(operador):
    for contador in range(1, 11):
        print(f"\nTabuada do {contador}:")

        for i in range(1, 11):
            if operador == 1:
                print(f"{contador} + {i} = {contador + i}")  # SOMA
            elif operador == 2:
                print(f"{contador} - {i} = {contador - i}")  # SUBTRAÇÃO
            elif operador == 3:
                print(f"{contador} X {i} = {contador * i}")  # MULTIPLICAÇÃO
            elif operador == 4:
                if i != 0:
                    print(f"{contador} / {i} = {contador / i}")  # DIVISÃO
                else:
                    print(f"Divisão por zero não é permitida para {contador} / {i}")
            else:
                print("Opção inválida, tente novamente!")

    return 0

def main():
    opcao = 1

    while opcao != 5:
        print("Digite 1 para adição.")
        print("Digite 2 para subtração.")
        print("Digite 3 para multiplicação.")
        print("Digite 4 para divisão.")
        print("Digite 5 para sair.")

        opcao = int(input())
        if opcao == 5:
            print("Obrigado, até a próxima!")
        else:
            calcula(opcao)

if __name__ == "__main__":
    main()

File: c/questao11.c
#include <stdio.h>


int calcula(int operador) {
    switch(operador){
        case 1: // SOMA
        for(int contador = 1; contador <= 10; contador++){
            printf("\nTabuada do %d\n:", contador);
            for(int i = 1; i <= 10; i++) {
                printf("%d + %d = %d\n", contador, i, contador + i);
            }
        }
        break;
        case 2: // SUBTRAÇÃO
        for(int contador = 1; contador <= 10; contador++){
            printf("\nTabuada do %d\n:", contador);
            for(int i = 1; i <= 10; i++) {
                printf("%d - %d = %d\n", contador, i, contador - i);
            }
        }
        break;
        case 3: // MULTIPLICAÇÃO
        for(int contador = 1; contador <= 10; contador++){
            printf("\nTabuada do %d\n:", contador);
            for(int i = 1; i <= 10; i++) {
                printf("%d X %d = %d\n", contador, i, contador * i);
            }
        }
        break;
        case 4: // DIVISÃO
        for(int contador = 1; contador <= 10; contador++){
            printf("\nTabuada do %d\n:", contador);
            for(int i = 1; i <= 10; i++) {
                printf("%d / %d = %d\n", contador, i, contador / i);
            }
        }
        break;
        default:
        printf("Opção inválida, tente novamente!\n");
    }
    return 0;
}
int main()
{
    int opcao = 1;
   
    while (opcao != 5){
        printf("Digite 1 para adição.\n");
        printf("Digite 2 para subtração.\n");
        printf("Digite 3 para multiplicação.\n");
        printf("Digite 4 para divisão.\n");
        printf("Digite 5 para sair.\n");
       
        scanf("%d", &opcao);
        if (opcao == 5){
            printf("Obrigado, até a próxima!\n");
        } else{
            calcula(opcao);
        }
    }

    return 0;
}